src/Table/Column/Tooltip.php

```diff
@@ -72,7 +72,7 @@ public function getHtmlTags(Model $row, ?Field $field): array
 
         return [
             '_' . $field->shortName . '_data_visible_class' => '',
-            '_' . $field->shortName . '_data_tooltip' => $tooltip,
+            '_' . $field->shortName . '_data_tooltip' => $this->getApp()->uiPersistence->typecastSaveField($row->getModel()->getField($this->tooltipField), $tooltip),
```

Comment:
Provide a proof you changes are consistent:

1. typecasting in all other places is done after/before? the if statement.
2. what is the 2nd field param?
3. how this change can be tested

Reply:
@mvorisek If you have an improvement to the submitted code, your submission would be highly welcomed. I cannot find another comparable getHtmlTags code as all others use getTag. At all other places, the if statement is based on non-typecasted values. It is obvious that the non-typecasted value will create issues, as this was missed to be corrected when type-strictness was increased. So happy if we can work together to solve this bug.

Reply:
Small tip for repro - you just add a decorator to any table:
```
         $table->addDecorator('any_field', new  \Atk4\Ui\Table\Column\Tooltip(
            [
                'tooltipField' => 'any_float_type_field',
                'icon' => 'bookmark red',
            ]));
```